Framework/API/test/CommonBinsValidatorTest.h

```diff
@@ -0,0 +1,60 @@
+#ifndef MANTID_API_COMMONBINSVALIDATORTEST_H_
+#define MANTID_API_COMMONBINSVALIDATORTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidAPI/CommonBinsValidator.h"
+#include "MantidTestHelpers/FakeObjects.h"
+
+using Mantid::API::CommonBinsValidator;
+
+class CommonBinsValidatorTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static CommonBinsValidatorTest *createSuite() {
+    return new CommonBinsValidatorTest();
+  }
+  static void destroySuite(CommonBinsValidatorTest *suite) { delete suite; }
+
+  void test_empty() {
+    auto ws = boost::make_shared<WorkspaceTester>();
+    CommonBinsValidator validator;
+    TS_ASSERT_EQUALS(validator.isValid(ws), "");
+  }
+
+  void xtest_zero_length_bins() {
```

Comment:
Is this test disabled (`xtest`) for any particular reason. I tried to run it and apparently it passes as it should.